native/NativeSampleApps/MobileSyncCompose/src/main/java/com/salesforce/samples/mobilesynccompose/contacts/ContactsActivity.kt

```diff
@@ -0,0 +1,133 @@
+import com.salesforce.samples.mobilesynccompose.core.ui.theme.SalesforceMobileSDKAndroidTheme
+import com.salesforce.samples.mobilesynccompose.core.ui.toWindowSizeRestrictions
+import com.salesforce.samples.mobilesynccompose.model.contacts.DefaultContactsRepo
+
+class ContactsActivity : ComponentActivity(), SalesforceActivityInterface {
+
+    private lateinit var vm: ContactsActivityViewModel
+    private lateinit var salesforceActivityDelegate: SalesforceActivityDelegate
+
+    @Suppress("UNCHECKED_CAST")
+    private val vmFactory = object : ViewModelProvider.Factory {
+        override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            // TODO Use Hilt to inject this
+            return DefaultContactsActivityViewModel(
+                DefaultContactsRepo(
+                    MobileSyncSDKManager.getInstance().userAccountManager.currentUser
+                )
+            ) as T
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        vm = ViewModelProvider(this, vmFactory)
+            .get(DefaultContactsActivityViewModel::class.java)
+
+        salesforceActivityDelegate = SalesforceActivityDelegate(this).also { it.onCreate() }
+
+        setContent {
+            /* We use the fact that LocalConfiguration is updated whenever configuration changes are
+             * detected to drive the layout restrictions recomposition: */
+            val windowSize = remember(LocalConfiguration.current) {
+                WindowMetricsCalculator.getOrCreate()
+                    .computeCurrentWindowMetrics(this)
+                    .bounds
+                    .toComposeRect()
+                    .size
+            }
+
+            val windowSizeRestrictions = with(LocalDensity.current) {
+                windowSize.toDpSize().toWindowSizeRestrictions()
+            }
+
+            SalesforceMobileSDKAndroidTheme {
+                ContactsActivityContent(
+                    layoutRestrictions = LayoutRestrictions(windowSizeRestrictions),
+                    vm = vm,
+                    onInspectDbClick = this::inspectDbClicked,
+                    onLogoutClick = this::logoutClicked,
+                    onSwitchUserClick = this::switchUserClicked,
+                    onSyncClick = this::syncClicked
+                )
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        salesforceActivityDelegate.onResume(true)
+    }
+
+    override fun onDestroy() {
+        salesforceActivityDelegate.onDestroy()
+        super.onDestroy()
+    }
+
+    override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
+        return salesforceActivityDelegate.onKeyUp(keyCode, event) || super.onKeyUp(keyCode, event)
+    }
+
+    override fun onResume(client: RestClient?) {
+        vm.sync(syncDownOnly = true)
+    }
+
+    override fun onLogoutComplete() {
```

Comment:
Remove - NO-OP.

Reply:
Oh that's interesting, is that a policy of ours?  I find empty method bodies without explicitly commenting them as no-ops confusing.